src/main/java/cms/gov/madie/measure/services/MeasureSetService.java

```diff
@@ -323,4 +304,51 @@ public List<Measure> getRecentMeasuresByMeasureSetId(List<String> measureSetIds)
     }
     return mostRecentMeasures;
   }
+
+  /**
+   * Change ownership in MeasureSet
+   *
+   * @param measureSetId - the MeasureSet that needs change of ownership
+   * @param userId - new owner
+   * @param retainShareAccess - add SHARED_WITH for the original owner if true, otherwise keep the
+   *     current Acls
+   * @param conductedBy - the user that performs the ownership change
+   * @return
+   */
+  public MeasureSet changeOwnership(
+      String measureSetId, String userId, boolean retainShareAccess, String conductedBy) {
+    Optional<MeasureSet> optionalMeasureSet = measureSetRepository.findByMeasureSetId(measureSetId);
+    if (optionalMeasureSet.isPresent()) {
+      MeasureSet measureSet = optionalMeasureSet.get();
+      String originalOwner = optionalMeasureSet.get().getOwner();
+      measureSet.setOwner(userId);
+      if (retainShareAccess) {
+        List<AclSpecification> acls =
+            !CollectionUtils.isEmpty(measureSet.getAcls())
+                ? measureSet.getAcls()
+                : new ArrayList<>();
+        acls.add(
+            AclSpecification.builder()
+                .userId(originalOwner)
+                .roles(Set.of(RoleEnum.SHARED_WITH))
+                .build());
+        measureSet.setAcls(acls);
+      }
+      MeasureSet updatedMeasureSet = measureSetRepository.save(measureSet);
+      ActionType actionType =
+          "apiKey".equalsIgnoreCase(conductedBy)
+              ? ActionType.UPDATED
+              : ActionType.OWNERSHIP_TRANSFER;
```

Comment:
@nmorasb, @jkotanchik-SB  can we just call it `OWNERSHIP_TRANSFER` in either case(whether it was performed by admin or user). in case of admin we set conductedBy=Admin or MADiE Admin?

Reply:
Sorry, Cecilia! more work for you which wasn't part of original ticket. rest looks good to me.

Reply:
@adongare @nmorasb This is done, I used "Admin" to replace "apiKey", let me know if it's not ok. Thanks!